interfaces/cython/cantera/examples/reactors/ic_engine.py

```diff
@@ -54,200 +56,203 @@
-            sim.set_max_time_step(1e-6 * 10. ** -n2)
-            n_last_refinement = n1
-    if not solved:
-        raise ct.CanteraError('Refinement limit reached')
-    # coarsen time step if too long ago
-    if n1 - n_last_refinement == n_wait_coarsening:
-        sim.set_max_time_step(1e-5)
-
-    # write output data
-    states.append(r.thermo.state)
-    V[n1] = r.volume
-    m[n1] = r.mass
-    mdot_in[n1] = inlet_valve.mdot(0)
-    mdot_out[n1] = outlet_valve.mdot(0)
-    d_W_v_d_t[n1] = - (r.thermo.P - ambient_air.thermo.P) * A_piston * \
-        piston_speed(t_i)
-    heat_release_rate[n1] = - r.volume * ct.gas_constant * r.T * \
-        np.sum(gas.standard_enthalpies_RT * r.thermo.net_production_rates, 0)
-
-
-#####################################################################
+states = ct.SolutionArray(cyl.thermo,
+                          extra=('t', 'ca', 'V', 'm',
+                                 'mdot_in', 'mdot_out',
+                                 'dWv_dt', 'heat_release_rate'))
+
+# simulate with a minimum resolution of 1 deg crank angle
+dt = 1. / (360 * f)
+t_stop = sim_n_revolutions / f
+while sim.time < t_stop:
+
+    # perform time integration
+    sim.advance(sim.time + dt)
+
+    # calculate results to be stored
+    dWv_dt = - (cyl.thermo.P - ambient_air.thermo.P) * A_piston * \
+        piston_speed(sim.time)
+    heat_release_rate = - cyl.volume * ct.gas_constant * cyl.T * \
+        np.sum(gas.standard_enthalpies_RT * cyl.thermo.net_production_rates, 0)
+
+    # append output data
+    states.append(cyl.thermo.state,
+                  t=sim.time*1e3, ca=crank_angle(sim.time),
+                  V=cyl.volume, m=cyl.mass,
+                  mdot_in=inlet_valve.mdot(sim.time),
+                  mdot_out=outlet_valve.mdot(sim.time),
+                  dWv_dt=dWv_dt,
+                  heat_release_rate=heat_release_rate)
+
+t = 1.e-3 * states.t
+
+# #####################################################################
 # Plot Results in matplotlib
-#####################################################################
+# #####################################################################
+
+
+def ca_ticks(t): return np.round(crank_angle(t) * 180 / np.pi, decimals=1)
 
-import matplotlib.pyplot as plt
 
 # pressure and temperature
-plt.clf()
-plt.subplot(211)
-plt.plot(t, states.P / 1.e5)
-plt.ylabel('$p$ [bar]')
-plt.xlabel('$\phi$ [deg]')
-plt.xticks(plt.xticks()[0], [])
-plt.subplot(212)
-plt.plot(t, states.T)
-plt.ylabel('$T$ [K]')
-plt.xlabel('$\phi$ [deg]')
-plt.xticks(plt.xticks()[0], crank_angle(plt.xticks()[0]) * 180 / np.pi,
-           rotation=17)
+fig, ax = plt.subplots(nrows=2)
+ax[0].plot(t, states.P / 1.e5)
+ax[0].set_ylabel('$p$ [bar]')
+ax[0].set_xlabel('$\phi$ [deg]')
+ax[0].set_xticklabels([])
+ax[1].plot(t, states.T)
+ax[1].set_ylabel('$T$ [K]')
+ax[1].set_xlabel('$\phi$ [deg]')
+ax[1].set_xticklabels(ca_ticks(ax[1].get_xticks()))
+fig.savefig('ic_engine_t_p_T.png')
```

Comment:
I know that this was the previous behavior of the script, but I'm not a fan of polluting the working directory when the example is run. For one, it prevents a clean run in, say, the CI service. I don't have a strong opinion that this line could be removed, but let's say its a mild preference 😸

Reply:
I share that sentiment, but didn't want to deviate too much from the original.